source/shared/core_conn.cpp

```diff
@@ -1005,6 +1074,41 @@ void column_encryption_set_func::func( _In_ connection_option const* option, _In
     conn_str += ";";
 }
 
+void ce_akv_str_set_func::func( _In_ connection_option const* option, _In_ zval* value, _Inout_ sqlsrv_conn* conn, _Inout_ std::string& conn_str TSRMLS_DC )
+{
+    SQLSRV_ASSERT( Z_TYPE_P( value ) == IS_STRING, "Azure Key Vault keywords accept only strings." );
+        
+    switch( option->conn_option_key )
+    {
+        case SQLSRV_CONN_OPTION_KEYSTORE_AUTHENTICATION: 
+        {
+            char *value_str = Z_STRVAL_P( value );        
+            CHECK_CUSTOM_ERROR( stricmp( value_str, "KeyVaultPassword" ) && stricmp( value_str, "KeyVaultClientSecret" ), conn, SQLSRV_ERROR_INVALID_AKV_AUTHENTICATION_OPTION )
```

Comment:
Should have checked the length of the string value here (instead of checking post connection), something like what we used to have for [KSP](https://github.com/Microsoft/msphpsql/blob/v5.2.0-RC/source/shared/core_conn.cpp?utf8=%E2%9C%93#L1077):

```
    size_t value_len = Z_STRLEN_P(value);

    CHECK_CUSTOM_ERROR(value_len == 0, conn, SQLSRV_ERROR_KEYSTORE_INVALID_VALUE) {
        throw core::CoreException();
    } 
```

Reply:
In addition, consider adding a member to indicate the valid AKV auth mode to the structure col_encryption_option, such that it is initialized to some INVALID value. Right here in this method you can set auth mode to either AKVCFG_AUTHMODE_PASSWORD or AKVCFG_AUTHMODE_CLIENTKEY. 

Thus no need to compare the values again in load_azure_key_vault()